wasi/wasi.go

```diff
@@ -107,6 +110,48 @@ func Stderr(writer io.Writer) Option {
 	}
 }
 
+// wasiStringArray holds null-terminated strings. It ensures that
+// its length and total buffer size don't exceed the max of uint32.
+type wasiStringArray struct {
+	strings      [][]byte
```

Comment:
rename to wasiArgs until it is reused somewhere besides args.

```suggestion
	// nullTerminatedValues are the UTF-8 encoded arg values with a NULL suffix for convenience writing to `argsBufPtr`
	nullTerminatedValues      [][]byte
```

Reply:
Changed the field's name as suggested. I'd like to keep `wasiStringArray` because I already know `environ_get` uses it. (https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-environ_getenviron-pointerpointeru8-environ_buf-pointeru8---result-errno) wdyt?

Reply:
I'm not sure internal types need to have the same name as a spec especially when this isn't an array, but I won't fuss over it. Usually specs should try not to micro-manage internal details and leave it to naming the interface and butting out otherwise